walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/MeetingRequest.java

```diff
@@ -46,6 +46,9 @@ public MeetingRequest(Collection<String> attendees, long duration) {
    * Returns a read-only copy of the people who are optional to attend this meeting.
    */
   public Collection<String> getOptionalAttendees() {
+    if (optional_attendees == null) {
+      return Collections.unmodifiableCollection(new HashSet<>());
+    }
```

Comment:
Oh yeah, we were only supposed to make changes to the query function, but when I tried to run the program on the local server I kept getting an error. I asked in the technical chat and others all had the same error, and they said to fix it by adding this (checking if null). I can review this and change it - I had just wanted to make sure that the function I was supposed to fill out was working properly.

Reply:
Okay, I think the error was because the constructor was being called, but optional_attendees isn't in the constructor. I changed it so that it is initialized to a new HashSet in the constructor and there's no error. Hopefully this is okay?

Reply:
Interesting, I don't see how that would have fixed the issue. Initializing the instance variable at declaration vs. in the constructor should have the same result.

Reply:
oh man, I meant to remove that part. That makes perfect sense what you are saying. Sorry for my mess up and re-requesting review too soon. I will continue to see if I can understand what the problem is!

Reply:
@hjohns28 out of curiosity, could you copy/paste the full error message you got before putting in this fix?

Reply:
Thanks for the error message. I'll LGTM this PR because the issue isn't within your own code. You can leave the null check in so that you don't get that error.

I'm guessing the error is the following:
The `MeetingRequest` class defined in script.js doesn't have the `optional_attendees` property defined (it only has `duration` and `attendees`). When that information is sent to the `QueryServlet` it is converted to a `MeetingRequest` defined by MeetingRequest.java using `gson.fromJson` (not the constructor). Since the Javascript `MeetingRequest` didn't have `optional_attendees` initialized, when it is converted to the Java `MeetingRequest` class, `optional_attendees` is null.

Happy to chat about this further if that's not clear.

Reply:
That makes sense - I was wondering about the MeetingRequest class in the script.js file. I might look at the project a little more, but I understand what you're saying and think that's likely. Thank you for the explanation and help, I appreciate it!